opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/OnConfigurationChangedListener.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opengrok.indexer.configuration;
+
+public interface OnConfigurationChangedListener {
+
+    void onConfigurationChanged();
```

Comment:
I'm thinking if it's a good idea to have the configuration as the parameter. Perhaps not as RE provides the locking mechanism for reads and writes.

Reply:
If `Configuration` could be accessed then yes but the RE is there for the locking.